tap_gitlab/__init__.py

```diff
@@ -218,27 +251,35 @@ def sync_project(pid):
             singer.write_state(STATE)
 
 
-def do_sync(pids):
+def do_sync():
     LOGGER.info("Starting sync")
 
+    gids = list(filter(None, CONFIG['groups'].split(' ')))
+    pids = list(filter(None, CONFIG['projects'].split(' ')))
+
     for resource, config in RESOURCES.items():
         singer.write_schema(resource, config['schema'], config['key_properties'])
 
-    for pid in pids:
-        sync_project(pid)
+    for gid in gids:
+        sync_group(gid, pids)
+
+    if not gids:
```

Comment:
Would you ever want to sync both groups and projects? It should be safe to not guard this for loop, right?

Reply:
Yes, please see my detailed explanation in the comment below or the README file.